pkg/ebpf/processor_funcs.go

```diff
@@ -366,32 +373,29 @@ func (t *Tracee) normalizeEventArgTime(event *trace.Event, argName string) error
 }
 
 // addHashArg calculate file hash (in a best-effort efficiency manner) and add it as an argument
-func (t *Tracee) addHashArg(event *trace.Event, fileName string, ctime int64) error {
-	if t.config.Output.CalcHashes {
-		// Currently Tracee does not support hash calculation of memfd files
-		if strings.HasPrefix(fileName, "memfd") {
-			return nil
-		}
-		hash, err := t.getFileHash(fileName, ctime, event.MountNS, event.ContainerID)
-
-		event.Args = append(
-			event.Args, trace.Argument{
-				ArgMeta: trace.ArgMeta{Name: "sha256", Type: "const char*"},
-				Value:   hash,
-			},
-		)
-		event.ArgsNum++
+func (t *Tracee) addHashArg(event *trace.Event, fileId *fileUniqueId, fileName string, ctime int64) error {
+	// Currently Tracee does not support hash calculation of memfd files
+	if strings.HasPrefix(fileName, "memfd") {
+		return nil
+	}
+	hash, err := t.getFileHash(fileId, fileName, ctime, event.MountNS, event.ContainerID)
 
-		// Container FS unreachable can happen because of race condition on any system,
-		// so there is no reason to return an error on it
-		if errors.Is(err, containers.ErrContainerFSUnreachable) {
-			logger.Debugw("failed to calculate hash", "error", err, "mount NS", event.MountNS)
-			err = nil
-		}
+	event.Args = append(
```

Comment:
Shouldn't you error check first?

Reply:
It was just a removal of outer check, but will adjust that. Thanks.

Reply:
You don't really need to check, as either way you want to add  the hash arg.
The only difference is that if it fails, the value will be empty.
Though, it might be good idea to put instead of an empty string of an hash an `nil` value if it fails🤔

Reply:
I agree. `nil` as we have for other args.